android/src/main/java/com/pauldemarco/flutterblue/FlutterBluePlugin.java

```diff
@@ -835,4 +843,12 @@ public void run() {
                 });
     }
 
+    private void exchangeGattMtu(int mtu, BluetoothGatt gatt) {
+        int retry = 5;
+        boolean status = false;
+        while (!status && retry > 0) {
+            status = gatt.requestMtu(mtu);
+            retry--;
```

Comment:
number of retries seem random.

Why retrying? 
Why 5 times?
Why to 512?

Seems very specific to a case you've to handle. Make sure this is a bit more generic, so that other people that don't need 512 (i.e. for the sake of performance), can do this also.

Reply:
Sorry, this has been done already:
#251

Reply:
@pauldemarco can be closed!